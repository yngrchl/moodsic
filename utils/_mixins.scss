
@import './_breakpoints.scss';

// Small devices
@mixin at-sm {
   @media (min-width: #{$screen-sm-min}) {
       @content;
   }
}

// Medium devices
@mixin at-md {
   @media (min-width: #{$screen-md-min}) {
       @content;
   }
}

// Large devices
@mixin at-lg {
   @media (min-width: #{$screen-lg-min}) {
       @content;
   }
}

// Extra large devices
@mixin at-xl {
   @media (min-width: #{$screen-xl-min}) {
       @content;
   }
}