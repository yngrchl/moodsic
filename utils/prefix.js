/**
 * Fix path for images
 * so they render even when deployed
 */

const prefix = process.env.NEXT_PUBLIC_BASE_PATH || '';

export { prefix };