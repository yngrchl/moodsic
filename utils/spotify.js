/**
 * Build Spotify login URL
 */
const authEndpoint = 'https://accounts.spotify.com/authorize';
const redirectUri = process.env.NEXT_PUBLIC_SPOTIFY_REDIRECT_URI;
const clientId = process.env.NEXT_PUBLIC_SPOTIFY_CLIENT_ID;

const scopes = [
  'user-read-private',
  'user-read-email'
];

export const loginUrl = `${authEndpoint}?client_id=${clientId}&response_type=code&redirect_uri=${redirectUri}&show_dialog=true&scope=${scopes.join(
  '%20'
)}`;
