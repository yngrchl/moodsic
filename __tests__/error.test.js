import React from 'react';
import { render, screen } from '@testing-library/react';
import Error from '../pages/error';
import { prefixPath } from '../utils/prefix';

describe('Error', () => {
  it('renders a heading', () => {
    render(<Error />);

    const heading = screen.getByRole('heading', {
      name: /oops, something went wrong/i,
    });

    expect(heading).toBeInTheDocument();
  });

  it('renders a try again button', () => {
    render(<Error />);

    const button = screen.getByText('TRY AGAIN FROM THE BEGINNING').closest('a');

    const path = prefixPath('/');
    const formattedPath = path === '/' ? path : path.replace(/\/$/, '');

    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('href', formattedPath);
  });
});
