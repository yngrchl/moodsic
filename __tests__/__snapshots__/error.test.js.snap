// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders Error page unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <main>
      <h1
        class="title"
      >
        <div
          class="image"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27150%27%20height=%27150%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="error icon of thunderstorm cloud"
              class="image"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        oops, something went wrong.
      </h1>
      <div
        style="margin-top: 5rem;"
      >
        <a
          class="btn error"
          href="/"
        >
          <h4>
            TRY AGAIN FROM THE BEGINNING
          </h4>
        </a>
      </div>
    </main>
    <footer>
      <p>
        source code at
      </p>
       
      <a
        href="https://github.com/yngrchl/moodsic"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Github logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </a>
    </footer>
  </div>
</div>
`;
